validation/src/test/scala/hmda/validation/rules/lar/validity/V338Spec.scala

```diff
@@ -0,0 +1,51 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V338Spec extends LarEditCheckSpec {
+  property("All applicants not meeting the criteria must pass") {
+    forAll(larGen) { lar =>
+      whenever(applicantNotOk(lar)) {
+        lar.mustPass
+      }
+    }
+  }
+
+  val okAction = List(1, 2, 3, 4, 5, 7, 8)
+  val actionTakenGen: Gen[Int] = Gen.oneOf(okAction)
+
+  property("Applicant meeting the criteria with the correct income must pass") {
+    forAll(larGen, actionTakenGen) { (lar: LoanApplicationRegister, x: Int) =>
+      val applicant = lar.applicant.copy(ethnicity = 4, race1 = 7, sex = 4, coEthnicity = 5,
+        coRace1 = 8, coSex = 5, income = "NA")
+      val validLar = lar.copy(applicant = applicant, actionTakenType = x)
+      validLar.mustPass
+    }
+  }
+
+  val incomeGen: Gen[Int] = Gen.choose(Int.MinValue, Int.MaxValue)
+
+  property("Applicant meeting the criteria with an incorrect income must fail") {
+    forAll(larGen, actionTakenGen, incomeGen) { (lar: LoanApplicationRegister, x: Int, income: Int) =>
+      val applicant = lar.applicant.copy(ethnicity = 4, race1 = 7, sex = 4, coEthnicity = 5,
+        coRace1 = 8, coSex = 5, income = income.toString)
+      val invalidLar = lar.copy(applicant = applicant, actionTakenType = x)
+      invalidLar.mustFail
+    }
+  }
+
+  override def check: EditCheck[LoanApplicationRegister] = V338
+
+  private def applicantNotOk(lar: LoanApplicationRegister): Boolean = {
```

Comment:
I don't buy this approach.  I changed an `and` to an `or` in the code under test, and then ran the test 5 times.  It failed 2 out of 5 times, which is better than 0, but not as good as 5.  I'm not sure what to suggest instead.  The obvious suggestion would be 7 different tests, but that seems like a lot of code.  When I think about it, it leads quickly to deeper questions of what we're trying to do and what we're trying to test.  (It's a rule.  Does the test need to be an [ugly mirror](http://jasonrudolph.com/blog/2008/07/30/testing-anti-patterns-the-ugly-mirror/) of the rule?  Would actual examples be better, instead of generated values?  That sort of question.)